src/spaceObjects/playerSpaceship.cpp

```diff
@@ -731,6 +731,11 @@ void PlayerSpaceship::applyTemplateValues()
     can_combat_maneuver = ship_template->can_combat_maneuver;
     can_self_destruct = ship_template->can_self_destruct;
     can_launch_probe = ship_template->can_launch_probe;
+    if (!on_new_player_ship_called)
```

Comment:
I think I likethis implementation the best. However, I would move this to the `onUpdate()` as that would ensure not only the template, but also other things like positions are applied.

Reply:
With the current location of on_new_player_ship it means the lua script is called before the playerShip can be seen by anywhere else in the lua code.

This in turn allows one to attach lua variables in the onNewPlayershipShip and assume that every playerShip seen inside of the update loop has got them attached. This in turn allows one to remove a pattern of checking those varibles for nil before use in the lua scripts.

If instead the on_new_player_ship is moved to the update loop then the lua code will have seen the playerShip once before it is updated.

on_new_player_ship could be called in both the shipSelectionScreen and objectCreationView but scripted playerSpaceships would still suffer from that (and it would be somewhat inconsistant)

with this in mind do you want to
1) keep as is
2) move on_new_player_ship into update
3) move on_new_player_ship into update, manually call it in objectCreationView and shipSelectionScreen
4) other that I havent thought of

Personally I think 1 or 3 is best